planner/core/logical_plans.go

```diff
@@ -468,24 +468,31 @@ func (ds *DataSource) deriveIndexPathStats(path *accessPath) (bool, error) {
 	} else {
 		path.tableFilters = ds.pushedDownConds
 	}
-	corColInAccessConds := false
 	if path.eqCondCount == len(path.accessConds) {
-		access, remained := path.splitCorColAccessCondFromFilters()
-		path.accessConds = append(path.accessConds, access...)
+		accesses, remained := path.splitCorColAccessCondFromFilters()
+		path.accessConds = append(path.accessConds, accesses...)
 		path.tableFilters = remained
-		if len(access) > 0 {
-			corColInAccessConds = true
-		}
-	}
-	path.indexFilters, path.tableFilters = splitIndexFilterConditions(path.tableFilters, path.index.Columns, ds.tableInfo)
-	if corColInAccessConds {
-		idxHist, ok := ds.stats.HistColl.Indices[path.index.ID]
-		if ok && !ds.stats.HistColl.Pseudo {
-			path.countAfterAccess = idxHist.AvgCountPerNotNullValue(ds.statisticTable.Count)
-		} else {
+		if len(accesses) > 0 && ds.statisticTable.Pseudo {
 			path.countAfterAccess = ds.statisticTable.PseudoAvgCountPerValue()
+		} else {
+			for i := range accesses {
+				col := path.idxCols[path.eqCondCount+i]
+				hist, ok := ds.statisticTable.Columns[col.ID]
+				var ndv float64
+				if ok && hist.Count > 0 {
+					factor := float64(ds.statisticTable.Count) / float64(hist.Count)
```

Comment:
Why `ds.statisticTable.Count`? This factor does not consider the selectivity of previous index conditions.